crates/nu-command/src/math/utils.rs

```diff
@@ -94,3 +94,86 @@ pub fn calculate(
         )),
     }
 }
+
+// Utility functions to reduce boiler plate in the the
+// trigonometric math commands
+pub fn trig_value<F>(value: Value, use_degrees: Option<bool>, f: F) -> Value
+where
+    F: Fn(f64) -> f64,
+{
+    let (val, span) = match value {
+        Value::Int { val, span } => (val as f64, span),
+        Value::Float { val, span } => (val, span),
+        _ => unreachable!(),
+    };
+
+    let use_degrees = use_degrees.unwrap_or(false);
+
+    let val = if use_degrees { val.to_radians() } else { val };
+
+    Value::Float { val: f(val), span }
+}
+
+pub fn trig_error(other: Value, head: Span) -> Value {
+    Value::Error {
+        error: ShellError::UnsupportedInput(
+            format!(
+                "Only numeric values are supported, input type: {:?}",
+                other.get_type()
+            ),
+            other.span().unwrap_or(head),
+        ),
+    }
+}
+
+pub fn trig_value_interval<F>(value: Value, use_degrees: bool, name: &str, f: F) -> Value
```

Comment:
Maybe worth communicating what the exact interval is, where this function is useful.